linera-base/src/data_types.rs

```diff
@@ -91,6 +91,55 @@ impl fmt::Display for Timestamp {
     }
 }
 
+/// Sometimes we need a serialization that is different from the usual one and
+/// and for example preserves order.
+/// The {to/from}_custom_bytes has to be coherent with the Borrow trait.
+pub trait CustomSerialize {
+    /// Serializes the value
+    fn to_custom_bytes(&self) -> Result<Vec<u8>, bcs::Error>;
```

Comment:
Forcing `bcs::Error` is odd. Idiomatically, this should be `Self::Error` for some associated type `Error`.

Reply:
That part is now obsolete due to other works being merged in the linera-view.